spanner/value_test.go

```diff
@@ -1240,6 +1260,18 @@ func TestDecodeValue(t *testing.T) {
 	type CustomNullTime NullTime
 	type CustomNullDate NullDate
 	type CustomNullNumeric NullNumeric
+	type CustomNullJSON NullJSON
+
+	type Message struct {
+		Name string
+		Body string
+		Time int64
+	}
+	msg := Message{"Alice", "Hello", 1294706395881547000}
+	jsonStr := `{"Name":"Alice","Body":"Hello","Time":1294706395881547000}`
```

Comment:
Same as above: Also add tests for an empty JSON array that is decoded as a `NullJSON`, and a JSON string that contains a null value.